WcaOnRails/app/models/user.rb

```diff
@@ -329,6 +329,10 @@ def can_manage_competition?(competition)
     can_admin_results? || competition.organizers.include?(self) || competition.delegates.include?(self)
   end
 
+  def can_edit_registration?(registration)
```

Comment:
That makes sense.

Breaking all that code out to an Ability (Capability?) class/module could be the clean way to do it.